include/mbgl/platform/settings.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <mapbox/value.hpp>
+
+#include <memory>
+
+namespace mbgl {
+namespace platform {
+
+#define MAPBOX_SETTING(name, value) constexpr const char* name = "mapbox_" #value
+
+// The value for EXPERIMENTAL_THREAD_PRIORITY_* keys, must be a double.
+MAPBOX_SETTING(EXPERIMENTAL_THREAD_PRIORITY_WORKER, thread_priority_worker);
+MAPBOX_SETTING(EXPERIMENTAL_THREAD_PRIORITY_FILE, thread_priority_file);
+MAPBOX_SETTING(EXPERIMENTAL_THREAD_PRIORITY_NETWORK, thread_priority_network);
+MAPBOX_SETTING(EXPERIMENTAL_THREAD_PRIORITY_DATABASE, thread_priority_database);
+
+// Settings class provides non-persistent, in-process key-value storage.
+class Settings final {
+public:
+    // Returns singleton instance
+    static Settings& getInstance() noexcept;
+
+    // Sets setting value for a specified key.
+    void set(const std::string& key, mapbox::base::Value value) noexcept;
+
+    // Sets multiple setting values by merging current Settings object
+    // with provided `values` object.
+    void set(mapbox::base::ValueObject values) noexcept;
```

Comment:
are those setters supposed to be called only once per key, or values are mutable runtime? E.g thread priority won't be changed for the running threads.

Reply:
In general case, values are mutable at runtime. As the team agreed, initially, no change observer functionality would be provided, as it wasn't requested.

For thread configuration specifically, client can set priorities for a map instance before it's creation. Once map is destroyed, new configuration can be set for a new instance.